src/fluent/utils.ts

```diff
@@ -18,17 +18,24 @@ const K8S_SA_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
 /**
  * Get the headers for a request
  *
+ * @param token - the token from @kubernetes/client-node
  * @returns the headers for undici
  */
-export async function getHeaders(): Promise<Record<string, string>> {
-  const token = await getToken();
+export async function getHeaders(token?: string | null): Promise<Record<string, string>> {
+  let saToken: string | null = "";
+  if (!token) {
+    saToken = await getToken();
```

Comment:
```suggestion
    svcAcctToken = await getToken();
```
I think svcAcct provides more clarity than sa when reading the function at a glance

Reply:
```bash
> k api-resources --namespaced=true | grep sa  
serviceaccounts             sa           v1                             true         ServiceAccount

> k create sa --help | egrep -A1 Aliases 
Aliases:
serviceaccount, sa
```

I would expect people that work on a Kubernetes Client to know that SA is serviceAccount but we can change it